Use c10::variant-based enums for F::grid_sample

yf225 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #29673 Use F::*FuncOptions for embedding/embeddingbag functionals
* #29584 Change arg dtype from float to double in LPPool and nn/utils/clip_grad.h
* #29536 Use c10::variant-based enums for SmoothL1Loss module and functional
* **#29535 Use c10::variant-based enums for F::grid_sample**
* #29404 Make all non-input arguments to functionals part of its options, and pass functional options by const-ref when possible

Differential Revision: [D18432273](https://our.internmc.facebook.com/intern/diff/D18432273)

Commit:
Use c10::variant-based enums for F::grid_sample

kostmo commented:
## CircleCI build failures summary
As of commit 1bd1b9f:
* **2/2** failures introduced in this PR
* **0/2** recognized as flaky



Here are the [reasons each build failed](https://dr.pytorch.org/commit-details.html?sha1=1bd1b9fd59e4a0ad30ccf760f0d89a6d7f1831bd).

---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

Commit:
Update on "Use c10::variant-based enums for F::grid_sample"

Use c10::variant-based enums for F::grid_sample

gh-metadata: pytorch pytorch 29535 gh/yf225/43/head

Commit:
Update on "Use c10::variant-based enums for F::grid_sample"

Use c10::variant-based enums for F::grid_sample

gh-metadata: pytorch pytorch 29535 gh/yf225/43/head

Commit:
Update on "Use c10::variant-based enums for F::grid_sample"

Use c10::variant-based enums for F::grid_sample

gh-metadata: pytorch pytorch 29535 gh/yf225/43/head
